### ISSUE TITLE
AttributeError: 'OrderedDict' object has no attribute 'startswith'

### ISSUE THREAD

**Issue by jungornti** [kennethreitz/pipenv#156]
Hello, during pipenv install --dev an exception occured:

```
pipenv install --dev
Creating a virtualenv for this project...
New python executable in /tmp/asd/st/.venv/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /tmp/asd/st/.venv
No package provided, installing all dependencies.
Fixing package names in Pipfile...
Pipfile found at /tmp/asd/st/Pipfile. Considering this to be the project home.
Pipfile.lock not found, creating...
Locking [dev-packages] dependencies...
Locking [packages] dependencies...
	Traceback (most recent call last):
  File "/usr/local/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pipenv/cli.py", line 669, in install
    do_init(dev=dev, requirements=requirements, allow_global=system)
  File "/usr/local/lib/python2.7/dist-packages/pipenv/cli.py", line 500, in do_init
    do_lock()
  File "/usr/local/lib/python2.7/dist-packages/pipenv/cli.py", line 390, in do_lock
    results = get_downloads_info(names_map, 'packages')
  File "/usr/local/lib/python2.7/dist-packages/pipenv/cli.py", line 352, in get_downloads_info
    if is_required_version(version, specified_version):
  File "/usr/local/lib/python2.7/dist-packages/pipenv/utils.py", line 138, in is_required_version
    if specified_version.startswith('=='):
AttributeError: 'OrderedDict' object has no attribute 'startswith'
```

The pipfile:
```
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true

[dev-packages]
django-debug-toolbar = "*"
pytest = "*"
pytest-django = "*"
coverage = "*"
tox = "*"
mock = "*"

[packages]
celery = "*"
django-allauth = "*"
django-ckeditor = "*"
django-cleanup = "*"
django-cmstemplates = "*"
django-codemirror-widget = "*"
django-el-pagination = "*"
django-widget-tweaks = "*"
easy-thumbnails = "*"
envvars = "*"
hiredis = "*"
ipython = "*"
lxml = "*"
psycopg2 = "*"
pymorphy2-dicts-ru = "*"
pytils = "*"
redis = "*"
rules = "*"
django = "*"
dj-database-url = "*"
pillow = "*"

[packages.pymorphy2]
extras = [ "fast",]
```

**Comment by jungornti** [comment]
It is failing on `packages.pymorphy2` section with extras requirements.

**Comment by FooBarQuaxx** [comment]
@kennethreitz Is this feature supported by Pipfile/pipenv?

**Comment by jungornti** [comment]
From the start my Pipfile was like this:

```
...
pymorphy = {extras: ["fast"]}
...
```

and after pipenv install it transformed to contain this section:

```
[packages.pymorphy2]
extras = [ "fast",]
```

**Comment by nateprewitt** [comment]
@jungornti, thanks for opening this. This was an oversight that I've got ready to patch but want to confirm a few things.

When did you use the `pymorphy = {extras: ["fast"]}` syntax in your Pipfile? I'm getting an exception from the TOML because this appears to be invalid syntax. Was this on an older version of pipenv?

Secondly, in the example provided it looks like you were trying to install `pymorphy` but instead had `pymorphy2` installed. That doesn't match your Pipfile above though, so I want to confirm that as a typo and not a bug.

**Comment by nateprewitt** [comment]
This is resolved with 6888b58, but I'm going to leave this open for a bit until @jungornti has a free moment to clarify those other issues.

Thanks again @jungornti :)

**Comment by asyncee** [comment]
Hello, Nate!

Check out [Pipfile examples](https://github.com/pypa/pipfile#examples), `extras` syntax is mentioned:

```
[packages]
requests = { extras = ['socks'] }
```

I know what @jungornti  was about, pymorphy and extras syntax mistakes were just typos (correct is `pymorphy2 = { extras = ['fast'] }`)

With this Pipfile i'm getting same error:

```
[[source]]
url = "https://pypi.python.org/simple"
verify_ssl = true

[packages]
pymorphy2-dicts-ru = "*"
pymorphy2 = {extras = ['fast']}
```

But this time Pipfile was not automatically transformed to contain separate `packages.pymorphy2` section.

Also, i confirm that this is fixed in commit [6888b58](https://github.com/kennethreitz/pipenv/commit/6888b580cd6496ed97f9918fa3a9b412a20660b2), have just tested it.

Thanks!

**Comment by nateprewitt** [comment]
Alright, I'm going to close this for now then. Feel free to reopen if you're still having issues @jungornti.

Thanks for testing things @asyncee!
